File: AndroidStudioProjects/Sample/app/src/main/java/com/example/sample/MainActivity.java
package com.example.sample;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private Button b ;
    private EditText name;
    private final int request = 10;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        b = findViewById(R.id.button1);
        name = findViewById(R.id.name1);

        b.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                String nameStr;
                nameStr = name.getText().toString();

                Intent intent = new Intent(MainActivity.this,SecondActivity.class);
                intent.putExtra("Message",nameStr);
                startActivityForResult(intent,request);
            }
        });
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if(requestCode == request){
            if(resultCode == RESULT_OK){
                String msg = data.getStringExtra("MS");
                b.setText(msg);
            }
        }else{
            b.setText("No output");
        }
    }
}

File: AndroidStudioProjects/ActivityDemo/app/src/main/java/com/example/activitydemo/MainActivity.kt
package com.example.activitydemo

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    private val TAG = "Life cycle"
    private lateinit var text:TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Log.i(TAG,"Main Activity onCreate called")

        val button = findViewById<Button>(R.id.Bt_next)
        var name:EditText = findViewById(R.id.Ed_name)
        text = findViewById(R.id.tv_saveInstance)

        //button.setOnClickListener(View.OnClickListener {  })
        button.setOnClickListener {
            val intent = Intent(this,SecondActivity::class.java)
            startActivity(intent)
        }
    }

    override fun onStart() {
        super.onStart()
        Log.i(TAG,"Main Activity onStart called")
    }

    override fun onResume() {
        super.onResume()
        Log.i(TAG,"Main Activity onResume called")
    }

    override fun onPause() {
        super.onPause()
        Log.i(TAG,"Main Activity onPause called")
    }

    override fun onStop() {
        super.onStop()
        Log.i(TAG,"Main Activity onStop called")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.i(TAG,"Main Activity onDestroy called")
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putInt("HighScore",321)
        Log.i(TAG,"Main Activity onSaveInstanceState called")
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)
        var content = savedInstanceState.getInt("HighScore")
        text.setText("$content")
        Log.i(TAG,"Main Activity onRestoreInstanceState called")
    }
}